distributed/bokeh/worker_monitor.py

```diff
@@ -60,6 +68,13 @@ def resource_append(lists, msg):
         lists[k].append(mean(pluck(k, L)) / 100)
 
     lists['time'].append(mean(pluck('time', L)) * 1000)
+    net = mean(pluck('network-send', L, 0))
+    if len(lists['time']) >= 2:
+        t1, t2 = tail(2, lists['time'])
```

Comment:
Ah, right.  Deques.  Using two integer indexes will still be much faster.  `t1, t2 = lists['time'][-2], lists['time'][-1]`

This deque can get long and this function is called very often.  This line makes the computation linear over a long list rather than constant time.